tests/pytests/test_index_oom.py

```diff
@@ -513,3 +512,490 @@ def test_oom_100_percent(env):
+    allShards_waitForIndexStatus(env, 'PAUSED_BEFORE_OOM_RETRY', idx)
+
+    # While paused, free memory so the retry can succeed
+    allShards_set_unlimited_maxmemory_for_oom(env)
+
+    # Resume again – indexing should now complete
+    run_command_on_all_shards(env,
+        f'{bgScanCommand()} SET_BG_INDEX_RESUME')
+    allShards_waitForIndexFinishScan(env, idx)
+
+    info = index_info(env, idx=idx)
+    assert info['num_docs'] == total_docs
+    assert to_dict(info['Index Errors'])['background indexing status'] == 'OK'
+    # Every shard’s failure counter must stay at 0
+    for shard_id in range(1, env.shardsCount + 1):
+        failures = env.getConnection(shard_id).execute_command(
+            'INFO', 'modules')['search_OOM_indexing_failures_indexes_count']
+        env.assertEqual(failures, 0)
+
+def test_pseudo_enterprise_cluster_oom_retry_failure(env):
+    verify_command_OK_on_all_shards(
+        env, '_FT.CONFIG SET _BG_INDEX_MEM_PCT_THR 80')
+    verify_command_OK_on_all_shards(
+        env, '_FT.CONFIG SET BG_INDEX_OOM_PAUSE_TIME 1')
+
+    conn = getConnectionByEnv(env)
+    docs_per_shard = 1_000
+    total_docs = docs_per_shard * env.shardsCount
+    for i in range(total_docs):
+        conn.execute_command('HSET', f'doc{i}', 'name', f'name{i}')
+
+    run_command_on_all_shards(env,
+        f'{bgScanCommand()} SET_PAUSE_ON_OOM true')
+    run_command_on_all_shards(env,
+        f'{bgScanCommand()} SET_PAUSE_ON_SCANNED_DOCS {docs_per_shard//4}')
+    run_command_on_all_shards(env,
+        f'{bgScanCommand()} SET_PAUSE_BEFORE_OOM_RETRY true')
+
+    idx = 'idx'
+    conn.execute_command('FT.CREATE', idx, 'SCHEMA', 'name', 'TEXT')
+
+    # Pause after first docs chunk, then tighten memory
+    allShards_waitForIndexPauseScan(env, idx)
+    allShards_set_tight_maxmemory_for_oom(env, 0.85)
+
+    # Resume – shards pause *before* OOM retry
+    run_command_on_all_shards(env,
+        f'{bgScanCommand()} SET_BG_INDEX_RESUME')
+    allShards_waitForIndexStatus(env, 'PAUSED_BEFORE_OOM_RETRY', idx)
+
+    # Resume again with memory still tight → PAUSED_ON_OOM
+    run_command_on_all_shards(env,
+        f'{bgScanCommand()} SET_BG_INDEX_RESUME')
+    allShards_waitForIndexStatus(env, 'PAUSED_ON_OOM', idx)
+
+    # One last resume – the second OOM turns into failure
+    run_command_on_all_shards(env,
+        f'{bgScanCommand()} SET_BG_INDEX_RESUME')
+    allShards_waitForIndexFinishScan(env, idx)
+
+    info = index_info(env, idx=idx)
+    errors = to_dict(info['Index Errors'])
+    env.assertEqual(errors['background indexing status'], 'OOM failure')
+    # Shards must report exactly one failed index each
+    for shard_id in range(1, env.shardsCount + 1):
+        failures = env.getConnection(shard_id).execute_command(
+            'INFO', 'modules')['search_OOM_indexing_failures_indexes_count']
+        env.assertEqual(failures, 1)
+
+@skip(cluster=True)
+def test_unlimited_memory_thrs(env):
```

Comment:
What is the purpose of this test? sanity?

Reply:
For coverage in `isBgIndexingMemoryOverLimit` function. 
`  if (RSGlobalConfig.indexingMemoryLimit == 0) {
    return false;
  }`